### PR TITLE
Sv 35/launch process

### PR DESCRIPTION
**Pull request by bjpirt**
# What?

Adds the page to kick off a new manual process. Due to the difficulty in working with the v1 MaT API in staging this has not been run against the actual API so I have created a ticket to do the integration (SV-166)

# Why?

So users can start new processes manually

# Checklist

- [x] Unit tests
- [-] Integration tests
- [x] Cypress tests

# Next steps

- [ ] Integration with the API when it is working in staging

### CODE REVIEW

**Review by LBHTKarki — COMMENTED** [review]
One property missing from the tmi object used in create task I think, but other than that looks really good and nice work working out the objects

File: src/gateways/v1MatAPIGateway.ts [hunk 1]
```diff
@@ -64,12 +73,37 @@ export default class v1MatAPIGateway implements v1MatAPIGatewayInterface {
...
+      });
+
+    return response;
+  }
+
+  public async getContactsByUprn(
+    uprn: string
+  ): Promise<GetContactsByUprnResponse> {
+    const response = await axios
+      .get(`${this.v1MatApiUrl}/v1/Contacts/GetContactsByUprn?urpn=${uprn}`, {
```
[the remark this answers is not in the document]
Comment (bjpirt) [reply src/gateways/v1MatAPIGateway.ts:98]: Good call

File: src/usecases/api/createManualTask.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    if (
+      (contacts && contacts.body && contacts.body.length === 0) ||
+      !contacts ||
+      !contacts.body
+    ) {
+      return { error: 404, body: undefined };
+    }
+
+    const contact = contacts.body[0];
+    const getUser = new GetUser(processData.officerEmail);
```
[the remark this answers is not in the document]
Comment (bjpirt) [reply src/usecases/api/createManualTask.ts:74]: 👍

File: src/usecases/api/createManualTask.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      enquirySubject: tmiLookup[processData.process].enquirySubject,
+      reasonForStartingProcess: processData.subProcess,
+      subject: 'c1f72d01-28dc-e711-8115-70106faa6a11',
+      natureofEnquiry: '15',
+      source: '1',
+      contactId: contact.contactId,
+      estateOfficerId: officerId.body, // Needs checking
+      estateOfficerName: processData.officerName,
+      officerPatchId: 'TBC', // Needs checking
+      areaName: 0, // Needs setting
```
[the remark this answers is not in the document]
Comment (LBHTKarki) [reply src/usecases/api/createManualTask.ts:88]: areaName is used to populate hackney_areaname field in TMI entity in CRM.
The field value (integer) must be one of the lookup values set for that field:

1 "Central"
2 "Clapton"
3 "Homerton 1"
4 "Homerton 2"
5 "Shoreditch"
6 "Stamford Hill"
7 "Stoke Newington"

File: src/usecases/api/createManualTask.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    ) {
+      return { error: 404, body: undefined };
+    }
+
+    const contact = contacts.body[0];
+    const getUser = new GetUser(processData.officerEmail);
+    const officerId = await getUser.execute();
+    if (!officerId.body) return { error: 404, body: undefined };
+
+    const tmi: TenancyManagementInteraction = {
```
[the remark this answers is not in the document]
Comment (bjpirt) [reply src/usecases/api/createManualTask.ts:78]: 👍

File: src/usecases/api/createManualTask.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    const contact = contacts.body[0];
+    const getUser = new GetUser(processData.officerEmail);
+    const officerId = await getUser.execute();
+    if (!officerId.body) return { error: 404, body: undefined };
+
+    const crmGateway = new CrmGateway();
+    const matPostgresGateway = new MatPostgresGateway();
+    const getOfficerPatchId = new GetOfficerPatch({
```
Comment (LBHTKarki) [inline src/usecases/api/createManualTask.ts:83]: this should be sufficient to get both officer and patch id, so the GetUser call above could be removed to simplify this a bit

Comment (bjpirt) [reply src/usecases/api/createManualTask.ts:83]: I'll do that - I wonder if we can combine these calls to simplify things a bit in the future

Comment (LBHTKarki) [reply src/usecases/api/createManualTask.ts:83]: yes, these were written separately, but are ultimately doing the same thing, so combining these in addition to the area details is perhaps the way to go 👍

**Pull request merged**